src/connection/xmpp/Connection.ts

```diff
@@ -7,20 +7,29 @@ import Account from '../../Account'
 import { AbstractConnection, Presence, STANZA_IQ_KEY, STANZA_KEY } from '../AbstractConnection'
 import XMPPJingleHandler from './JingleHandler'
 import Client from '../../Client'
+import Contact from '@src/Contact';
 
 export default class XMPPConnection extends AbstractConnection implements IConnection {
    private handler: XMPPHandler;
-
    constructor(account: Account, protected connection) {
       super(account);
-
       this.handler = new XMPPHandler(account, connection);
       this.handler.registerHandler();
       NS.register('METADATA_NOTIFY', 'urn:xmpp:avatar:metadata+notify');
 
+      this.getPEPService().subscribe('http://jabber.org/protocol/nick', nickHandler);
+
       this.processPendingStorageConnections();
       this.listenToStorageConnections();
 
+      function nickHandler(stanza): boolean {
```

Comment:
Inline function functions should be avoided.

Reply:
We tried defining the function outside the constructor, but this results in a TypeError because 'account' allegedly is undefined (although it still works). Is this solution preferred although it results in errors?

Reply:
Just fix the error :wink:. Test which value `this` has and use maybe an arrow function.

Reply:
Like this?